files/initramfs-tools/fsck-rootfs

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+case $1 in
+    prereqs)
+        exit 0
+        ;;
+esac
+
+# Extract kernel parameters
+root_val=""
+set -- $(cat /proc/cmdline)
+for x in "$@"; do
+    case "$x" in
+        root=*)
+            root_val="${x#root=}"
+            ;;
+    esac
+done
+
+# Check the filesystem we are using
+if [ ! -z $root_val ]; then
+    fstype=$(blkid -o value -s TYPE $root_val)
+    case "$fstype" in
+        ext4)
+            fsck.ext4 -v -p $root_val 2>&1 | gzip -c > /tmp/fsck.log.gz
+            ;;
+        ext3)
```

Comment:
>ext3 [](start = 8, length = 4)

You can combine the 2 branches into one, and share the same command with variable.

Reply:
I see your improvement, thanks! Is it possibel to make it even better. Not sure it works for the old `sh`

```bash
        ext4|ext3)
            cmd="fsck.$fstype -v -p"
            ;;
```

---
In reply to: [409184961](https://github.com/Azure/sonic-buildimage/pull/4431#discussion_r409184961) [](ancestors = 409184961)

Reply:
I tried your suggestion, but for some reason, that did not worked, I could not see any fsck logging.